src/lando/main/tests/__init__.py

```diff
@@ -66,3 +66,22 @@ def test_require_phabricator_api_key(monkeypatch, optional, valid_key, status):
         assert resp.body.api_token == "custom-key"
 
     assert resp.status_code == status
+
+
+@pytest.mark.django_db(transaction=True)
+def test_phabricator_api_key_encryption():
+    user = User.objects.create_user(username="test_user", password="test_password")
+    profile = Profile.objects.create(user=user)
+
+    assert profile.phabricator_api_key == ""
+
+    # Set an arbitrary key.
+    key = "test-key"
+    profile.save_phabricator_api_key(key)
+    assert profile.encrypted_phabricator_api_key
```

Comment:
>Fernet tokens are also base64 encoded, we could check to see if the encrypted key decodes/re-encodes cleanly.

We're already checking the original key against the decrypted key (line 82), should that not be covered already? Unless you are referring to something else.

Reply:
That check is ensuring the encryption/decryption itself is working as expected. I'm thinking more about testing that the encrypted key has the format we are expecting, base64 encoded bytes for Fermat. For example a test like:

```python
assert base64.encode(base64.decode(profile.encrypted_phabricator_api_key)) == profile.encrypted_phabricator_api_key
```

This way we can confirm that the value in `profile.encrypted_phabricator_api_key` is base64.

Reply:
>I'm thinking more about testing that the encrypted key has the format we are expecting

That feels like we are testing tangential functionality in the cryptography library, rather than testing something that we are implementing ourselves, which seems out of scope to me. I.e., nothing in our code is directly interfacing with `base64` or necessarily cares about the format of the stored encryption key as long as it can be encrypted/decrypted seamlessly.

Thoughts?

Reply:
I'm not too fussed about specifically testing base64, mostly just suggesting we should test to ensure the encrypted value is something we are expecting. ie the value is not empty, is an appropriate length, is base64 and not some other format, etc.

I'm mostly trying to avoid a situation where we make a change and without realizing it we are storing junk values in the DB, or storing unencrypted key content etc. I know it's unlikely to happen but if we can add a test now that does a few sanity checks of our implementation it would avoid headaches down the line. The tests we have now cover a few bases but I wanted to pick your brain about any other simple things we could add now.